### PR TITLE
Cleanup of type safe builders

### PR DESCRIPTION
**Pull request by bibryam**

### CODE REVIEW

**Comment by cloudbotinmarsat** [comment]
SonarQube analysis reported 2 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MINOR [RouteKubernetesModelProcessor.java#L7](https://github.com/Inmarsat-itcloudservices/contacts-example/blob/3e2e5f7630925ecb07cac6898043aafac5c6a089/src/main/java/com/inmarsat/demo/buildconfig/common/RouteKubernetesModelProcessor.java#L7): Remove this unused import 'io.fabric8.openshift.api.model.TemplateBuilder'. [rule](http://ec2-54-194-136-253.eu-west-1.compute.amazonaws.com:9000/coding_rules#rule_key=squid%3AUselessImportCheck)
1. MINOR [ServiceKubernetesModelProcessor.java#L34](https://github.com/Inmarsat-itcloudservices/contacts-example/blob/3e2e5f7630925ecb07cac6898043aafac5c6a089/src/main/java/com/inmarsat/demo/buildconfig/common/ServiceKubernetesModelProcessor.java#L34): Define a constant instead of duplicating this literal "contacts-example" 3 times. [rule](http://ec2-54-194-136-253.eu-west-1.compute.amazonaws.com:9000/coding_rules#rule_key=squid%3AS1192)

**Comment by cloudbotinmarsat** [comment]
SonarQube analysis reported 2 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MINOR [RouteKubernetesModelProcessor.java#L7](https://github.com/Inmarsat-itcloudservices/contacts-example/blob/51d450aa775796a6383efcf08a4e567a8306654a/src/main/java/com/inmarsat/demo/buildconfig/common/RouteKubernetesModelProcessor.java#L7): Remove this unused import 'io.fabric8.openshift.api.model.TemplateBuilder'. [rule](http://ec2-54-194-136-253.eu-west-1.compute.amazonaws.com:9000/coding_rules#rule_key=squid%3AUselessImportCheck)
1. MINOR [ServiceKubernetesModelProcessor.java#L34](https://github.com/Inmarsat-itcloudservices/contacts-example/blob/51d450aa775796a6383efcf08a4e567a8306654a/src/main/java/com/inmarsat/demo/buildconfig/common/ServiceKubernetesModelProcessor.java#L34): Define a constant instead of duplicating this literal "contacts-example" 3 times. [rule](http://ec2-54-194-136-253.eu-west-1.compute.amazonaws.com:9000/coding_rules#rule_key=squid%3AS1192)

File: environments/uat/configmap/app.properties [hunk 1]
```diff
@@ -0,0 +1 @@
+username=uat
```
Comment (cloudbotinmarsat) [inline environments/uat/configmap/app.properties]: INFO Remove this hard-coded username. [rule](http://ec2-54-194-136-253.eu-west-1.compute.amazonaws.com:9000/coding_rules#rule_key=jproperties%3AS2068)

**Comment by cloudbotinmarsat** [comment]
SonarQube analysis reported 5 issues
* MINOR 2 minor
* INFO 3 info

Watch the comments in this conversation to review them.

#### 2 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MINOR [RouteKubernetesModelProcessor.java#L7](https://github.com/Inmarsat-itcloudservices/contacts-example/blob/bf56a63ea9127b00cdcc4fa154d9aba594497b5f/src/main/java/com/inmarsat/demo/buildconfig/common/RouteKubernetesModelProcessor.java#L7): Remove this unused import 'io.fabric8.openshift.api.model.TemplateBuilder'. [rule](http://ec2-54-194-136-253.eu-west-1.compute.amazonaws.com:9000/coding_rules#rule_key=squid%3AUselessImportCheck)
1. MINOR [ServiceKubernetesModelProcessor.java#L34](https://github.com/Inmarsat-itcloudservices/contacts-example/blob/bf56a63ea9127b00cdcc4fa154d9aba594497b5f/src/main/java/com/inmarsat/demo/buildconfig/common/ServiceKubernetesModelProcessor.java#L34): Define a constant instead of duplicating this literal "contacts-example" 3 times. [rule](http://ec2-54-194-136-253.eu-west-1.compute.amazonaws.com:9000/coding_rules#rule_key=squid%3AS1192)

File: src/main/java/com/inmarsat/demo/ContactsExampleRoute.java [hunk 2]
```diff
@@ -29,19 +29,19 @@
  * Configures all our Camel routes, components, endpoints and beans
  */
 @ContextName("myJettyCamel")
-public class MyJettyRoute extends RouteBuilder {
+public class ContactsExampleRoute extends RouteBuilder {
 
     @Inject @Uri("jetty:http://0.0.0.0:8080/camel/hello")
```
Comment (cloudbotinmarsat) [inline src/main/java/com/inmarsat/demo/ContactsExampleRoute.java]: CRITICAL Make this IP "0.0.0.0" address configurable. [rule](http://ec2-54-194-136-253.eu-west-1.compute.amazonaws.com:9000/coding_rules#rule_key=squid%3AS1313)

**Comment by cloudbotinmarsat** [comment]
SonarQube analysis reported 5 issues
* CRITICAL 1 critical
* MINOR 4 minor

Watch the comments in this conversation to review them.

#### 4 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MINOR [ContactsExampleRouteTest.java#L21](https://github.com/Inmarsat-itcloudservices/contacts-example/blob/5fb45368fee320d658abe28732c70626d31d669b/src/test/java/com/inmarsat/demo/ContactsExampleRouteTest.java#L21): Define a constant instead of duplicating this literal "  <contact>\n" 3 times. [rule](http://ec2-54-194-136-253.eu-west-1.compute.amazonaws.com:9000/coding_rules#rule_key=squid%3AS1192)
1. MINOR [ContactsExampleRouteTest.java#L25](https://github.com/Inmarsat-itcloudservices/contacts-example/blob/5fb45368fee320d658abe28732c70626d31d669b/src/test/java/com/inmarsat/demo/ContactsExampleRouteTest.java#L25): Define a constant instead of duplicating this literal "  </contact>\n" 3 times. [rule](http://ec2-54-194-136-253.eu-west-1.compute.amazonaws.com:9000/coding_rules#rule_key=squid%3AS1192)
1. MINOR [ContactsExampleRouteTest.java#L55](https://github.com/Inmarsat-itcloudservices/contacts-example/blob/5fb45368fee320d658abe28732c70626d31d669b/src/test/java/com/inmarsat/demo/ContactsExampleRouteTest.java#L55): Define a constant instead of duplicating this literal "<contact>" 3 times. [rule](http://ec2-54-194-136-253.eu-west-1.compute.amazonaws.com:9000/coding_rules#rule_key=squid%3AS1192)
1. MINOR [ContactsExampleRouteTest.java#L59](https://github.com/Inmarsat-itcloudservices/contacts-example/blob/5fb45368fee320d658abe28732c70626d31d669b/src/test/java/com/inmarsat/demo/ContactsExampleRouteTest.java#L59): Define a constant instead of duplicating this literal "  </contact>" 3 times. [rule](http://ec2-54-194-136-253.eu-west-1.compute.amazonaws.com:9000/coding_rules#rule_key=squid%3AS1192)

**Comment by akeithball** [comment]
test this please

**Comment by akeithball** [comment]
test this please

**Comment by cloudbotinmarsat** [comment]
SonarQube analysis reported 4 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MINOR [ContactsExampleRouteTest.java#L21](https://github.com/Inmarsat-itcloudservices/contacts-example/blob/5fb45368fee320d658abe28732c70626d31d669b/src/test/java/com/inmarsat/demo/ContactsExampleRouteTest.java#L21): Define a constant instead of duplicating this literal "  <contact>\n" 3 times. [rule](http://ec2-54-194-136-253.eu-west-1.compute.amazonaws.com:9000/coding_rules#rule_key=squid%3AS1192)
1. MINOR [ContactsExampleRouteTest.java#L25](https://github.com/Inmarsat-itcloudservices/contacts-example/blob/5fb45368fee320d658abe28732c70626d31d669b/src/test/java/com/inmarsat/demo/ContactsExampleRouteTest.java#L25): Define a constant instead of duplicating this literal "  </contact>\n" 3 times. [rule](http://ec2-54-194-136-253.eu-west-1.compute.amazonaws.com:9000/coding_rules#rule_key=squid%3AS1192)
1. MINOR [ContactsExampleRouteTest.java#L55](https://github.com/Inmarsat-itcloudservices/contacts-example/blob/5fb45368fee320d658abe28732c70626d31d669b/src/test/java/com/inmarsat/demo/ContactsExampleRouteTest.java#L55): Define a constant instead of duplicating this literal "<contact>" 3 times. [rule](http://ec2-54-194-136-253.eu-west-1.compute.amazonaws.com:9000/coding_rules#rule_key=squid%3AS1192)
1. MINOR [ContactsExampleRouteTest.java#L59](https://github.com/Inmarsat-itcloudservices/contacts-example/blob/5fb45368fee320d658abe28732c70626d31d669b/src/test/java/com/inmarsat/demo/ContactsExampleRouteTest.java#L59): Define a constant instead of duplicating this literal "  </contact>" 3 times. [rule](http://ec2-54-194-136-253.eu-west-1.compute.amazonaws.com:9000/coding_rules#rule_key=squid%3AS1192)

**Pull request merged**